### ISSUE TITLE
Rework air defense framework

### ISSUE THREAD

**Issue by wrycu** [shdwp/dcs_liberation#68]
Taking several cues from @Khopa, I'd like to implement the following changes to air defenses:

* **Assign air defenses to locations**: Rather than having a particular base manage air defenses in it's inventory, we should assign specific air defenses to air defense sites at the start of the campaign.  While these units can be upgraded as the campaign progresses, they should otherwise retain the same unit composition regardless of the state of the nearby bases
* **Constant SAM presence**: Air defenses in a reasonable range of the mission being conducted should be spawned regardless of if they're part of the mission being performed or not. Note that this has been partially implemented on the dev branch already
* **Front-line air defenses**: Various mission types being conducted against the front-lines should contain additional air defenses, e.g. MANPADs and other short-range air defenses. Additionally, medium-value targets such as FARPs should have a chance of random point-defense sites
* **Imperfect information**: The exact number of enemy air defenses should not be presented in the briefing, though the types of these units is reasonable
* **IADS**: With a more static composition of air defenses, it may be possible to bring in the concept of [IADS](http://en.citizendium.org/wiki/Integrated_air_defense_system) in to the SAM sites, permitting more complex strategies to counter SEAD/DEAD (e.g. information being shared without the active usage of every radar at the same time)
* **ELINT**: I'm not sure if it's possible yet, but investigate the introduction of [ELINT](http://en.citizendium.org/wiki/Electronic_intelligence) as a source of enemy SAM unit composition
* **Configuration**: Some of these features may introduce unwanted components in certain mission types (e.g. make it harder to perform air intercept missions.) As such, there should be an option to turn at least some of them off

I'm open to other ideas and concepts to add to this.

**Comment by Khopa** [comment]
> Assign air defenses to locations: Rather than having a particular base manage air defenses in it's inventory

Yes, but i also think the airbase should retain a few units in it's inventory for it's own air defense. 

As an armchair general, here is what i think the algorithm should generate for instance : 

![image](https://user-images.githubusercontent.com/2546901/60761664-78bba480-a04d-11e9-824b-05e5fa39ff86.png)

**A) Airbase defense :**

We can't let an airbase totally undefended. An airbase is basically an air defense site, it must be defended with close and medium range system.

*Appropriate units :* Roland, Rapier, SA-13, SA-19, Zsu-23, 

**B) AA-Site :** 

They should be dedicated to long range air defense, one should at least cover the local airbase, as well as other potential targets, when possible. 

The aa-site would be made of one long range system as well as several close & medium range systems for it's own protection.

*Appropriate units :* Patriot, SA-10 (S-300), SA-3 (S-125), SA-2 (S-75)

**C) Strike target defense :** 

I think at least some of them should be covered with basic aa defense, such as AAA & Manpads.
And i believe it's already implemented.

*Appropriate units :* Stinger, Igla, Strella, Zu-23, 

**D) Frontline units :** 

Currently, on the frontline, we only have the armored units.
We could protect them with close range AAA, and also eventually mobile SAM systems (such as SA-6/8/11, Avenger, Roland, Tunguska). But this could make the mission very difficult, and AI CAS flight could be decimated.

Same for allied units.

**Note :** 

It's great on paper to generate a lot of units, but we must also be cautious. Will DCS engine support it well ? 

***

> Constant SAM presence

Yes

> IADS

I love this idea.

We could integrate into the mission file generated an "AI" lua script, that would turn on and off radars, based on units detected by the EWR radar (We'd need to add EWR site to the generator first)

The AI commander would also scramble interceptors when needed (and if available in nearby airbase), based on the available EWR data.

> ELINT

I love this idea as well, but no idea how to implement it.

> Imperfect information:

Could be a configurable difficulty setting : 

Intel Level :
* Blind : No idea what the enemy has
* Estimation : Estimation of the number of each unit type (Example : 8-12 tanks, 4-8 fighters, 0-1 SAM)
* Omniscient : You know about every units and get precise coordinates at mission start

In a future implementation, the "intel level" of the player could evolve based on his actions, using events such as succesful recon/elint flights, coms tower captured... etc

> Configuration: Some of these features may introduce unwanted components in certain mission types (e.g. make it harder to perform air intercept missions.) As such, there should be an option to turn at least some of them off

Agreed.  We currently have the "SAM" checkbox at the game startup to disable units that are hard to deal with.

**Comment by wrycu** [comment]
@Khopa has done some work on this:
f698cb66b8c50a1f017dfd1f3a06a20ea4e0b8d2
f5a5fb765b8ff1a4826d4981edc9687ff0779aee
6b77e1cce5737844da26bee4f7d31c8d130c29e8
09b8ff6b93dfdf39a56b49d703dbdaab777a90f0
